Automattic/jetpack
# Gutenberg: Add Top Posts and Pages Block

Fixes #32937

## Proposed changes:

Converts the legacy Top Posts & Pages widget into a Gutenberg block! It has the same features; you can still choose to filter out different types of media and customise the display, but the block also allows you to choose the timeframe of stats which to draw on. 

The only exception is that this block is based entirely on views/traffic. The widget allowed determining top content from Likes, but I don't have access to a WordPress.com environment that would allow me to add this, so I'm afraid that somebody else would need to handle that in future. I personally don't feel like that's a massive deal though, at least not for this first version. 

<img width="1580" alt="Screenshot 2023-11-17 at 23 34 23" src="https://github.com/Automattic/jetpack/assets/43215253/d15a1334-cbd3-4509-8858-5483d98cad77">

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


N/A

## Testing instructions:
I'm hoping that the block should be fairly intuitive to use! Just like the Related Posts block, you should be prompted to activate Stats if the module is disabled. In fact, I've drawn quite heavily on that block - for example, the metadata and layout settings are the same. 

<img width="761" alt="Screenshot 2023-11-18 at 15 26 04" src="https://github.com/Automattic/jetpack/assets/43215253/a867fff5-cd8f-43cd-a113-f2255681c7cb">

If Stats is active, some posts should automatically display. If there aren't enough posts within the given timeframe, then random posts should be used to fill the rest of the block. There are some settings to filter out content and to choose which metadata information to display, so definitely verify that those work. 

Ultimately though, it should render nicely in the frontend. 

<img width="697" alt="Screenshot 2023-11-18 at 15 28 42" src="https://github.com/Automattic/jetpack/assets/43215253/2264803f-63dd-44e8-aac8-36de236dd4b2">
